### DISCUSSION TITLE
How is the GitHub token injected into git?

### DISCUSSION THREAD

**Discussion by gaborcsardi** [community/community#25255] [Actions]
<p>As I understand, within a run command line <code>git</code> has access to the GitHub token that was created specifically for the run.</p>
<p>How is the token injected into git here? Is it added to the git credential store? Or some other way?</p>

**Comment by airtower-luna** [comment]
<p>Check the output of the <code>actions/checkout</code> step in your workflow log. Assuming it’s using Git to retrieve the repository (as opposed to the API if Git isn’t available on the runner) you’ll see it setting up an <code>Authorization</code> header in the “Setting up auth” section.</p>

**Comment by jsoref** [comment, marked as answer]
<p>No it isn’t done w/ the git credential store, it’s done w/ the <code>.git/config</code> file:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/git-auth-helper.ts#L55-L65">
  <header class="source">

      <a href="https://github.com/actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/git-auth-helper.ts#L55-L65" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/git-auth-helper.ts#L55-L65" target="_blank" rel="noopener">actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/git-auth-helper.ts#L55-L65</a></h4>



    <pre class="onebox">      <code class="lang-ts">
        <ol class="start lines" start="55" style="counter-reset: li-counter 54 ;">
            <li>this.tokenConfigKey = `http.${serverUrl.origin}/.extraheader` // "origin" is SCHEME://HOSTNAME[:PORT]</li>
            <li>const basicCredential = Buffer.from(</li>
            <li>  `x-access-token:${this.settings.authToken}`,</li>
            <li>  'utf8'</li>
            <li>).toString('base64')</li>
            <li>core.setSecret(basicCredential)</li>
            <li>this.tokenPlaceholderConfigValue = `AUTHORIZATION: basic ***`</li>
            <li>this.tokenConfigValue = `AUTHORIZATION: basic ${basicCredential}`</li>
            <li>
            </li>
<li>// Instead of SSH URL</li>
            <li>this.insteadOfKey = `url.${serverUrl.origin}/.insteadOf` // "origin" is SCHEME://HOSTNAME[:PORT]</li>
        </ol>
      </code>
    </pre>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
